test/fluree/db/query/misc_queries_test.clj

```diff
@@ -55,7 +55,13 @@
                                               "foo:x" {"@container" "@list"}}]
                               :where     [['?s "@id" "foo:dan"]]
                               :select    {'?s ["*"]}}))
-        "override unwrapping with @list")))
+        "override unwrapping with @list")
+    (is (= [{"@id"     "http://example.org/ns/dan"
+             "http://example.org/ns/x" 1}]
+           @(fluree/query db {"@context" ["" nil]
+                              :where     [['?s "@id" "http://example.org/ns/dan"]]
+                              :select    {'?s ["*"]}}))
```

Comment:
I think the most ideal syntax here to clear the default context might be:
```
{"@context" nil
 :where     [['?s "@id" "http://example.org/ns/dan"]]
 :select    {'?s ["*"]}}
```
I'd think of app too that might be using some code to generate contexts and these would ideally produce the same behavior:
```
{"@context" {}
 :where     [['?s "@id" "http://example.org/ns/dan"]]
 :select    {'?s ["*"]}}
```
and
```
{"@context" []
 :where     [['?s "@id" "http://example.org/ns/dan"]]
 :select    {'?s ["*"]}}
```

Reply:
The new `retrieve-context` function used by the Db `-context` method can't currently differentiate between no-context-found (i.e. no `@context` key) and an `@context` with a value of nil - it's currently set up so that a `nil` supplied context does the opposite of clearing the context, it falls back to the default context. 

Which means we'd need to pass some sort of flag to the `-context` method as the supplied context, in order to distinguish between `nil` (not found) and `nil` (clear context). Which feels a bit messy, but it could work. I'll put together a commit that enables the ideal api you outlined above.

Reply:
@dpetran have you had a chance to put the commit together you were describing? Are there any other action items needed before we can merge?

Reply:
I've updated this branch with the changes needed to enable the proposed syntax here.

Reply:
I think it is perfectly fine, BTW, to have people clear the default context with an empty map `{}` (and not support `nil`, if that creates additional issues). If we end up finding a lot of people try to do this with with `nil`, and we feel we've done what we can in our documentation to let people know `{}` is the proper syntax, then we could go through the hassle of supporting `nil` as well. I don't think it is worth building special conditions yet for something we don't know if is a problem.

So below can be the way we tell people to do this:

```
{"@context" {}
 :where     [['?s "@id" "http://example.org/ns/dan"]]
 :select    {'?s ["*"]}}
```

Reply:
I prefer this as well to simplify the underlying code. `nil` is a nice abstraction, but the hack to enable it probably isn't worth it since we have the `{}` alternative which should result in much simpler code without the need of a special keyword.

Reply:
Sorry @dpetran for the back and forth.

Reply:
Well, strictly speaking the hack of `nil` is the only standard way of clearing context, using `[]` and `{}` is us inventing some syntax to do it, fwiw. Especially if we want to properly handle arbitray json-ld that's coming in we'll need to distinguish a `nil` context from a missing context.

Reply:
> `nil` is the only standard way of clearing context

Sort of. I think this is apples and oranges, the use of `nil` like this is hard to claim as standard, as in standard JSON-LD usage, having `nil` as a top-level `@context` would never make sense. You'd only use `nil` in nested JSON-LD data structure to reset parent `@context` values in child-levels... and while we want to support that use, the purpose here is different.

Why do we need a unique way to express this? Because the spec is focused on published data, not on the nuances if someone (us) contemplated creating an operational JSON-LD database. So we'll have situations like this where we have to come up with our own solutions regardless.